assets/js/util.js

```diff
@@ -242,8 +243,26 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function apyCall(requestObject, endpoint) {
+    window[requestObject.requestName] = $.ajax({
```

Comment:
> The whole point of having an arbitrary options Object and using Object.assign is that the callee can pass in arguments callApy doesn't know.

What does arbitrary mean? It can have any random `key: value` pairs.
 So, it may happen that, my `options` object is simply a `{a: 1, b: 2}` and I may pass this to `callApy` as `options` argument? In this case, how do I set the `data`, `success` etc values in the `ajax` call?

One solution I thought of is to have my `options` object modified as the one in below snippet. I make use of the `keys` that are valid in an ajax call. Now, when `Object.assign` runs, it copies the contents of this object into the target and executes the function! 

```
callApy({
       'data': request,
       'success': handleTranslateSuccessResponse,
       'error': translationNotAvailable,
       'requestName': 'textTranslateRequest'
}, endpoint);

function callApy(options, endpoint) {
    var targetObj = Object.assign({
        url: config.APY_URL + endpoint,
        beforeSend: ajaxSend,
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
        dataType: 'json'
    }, options);

    window[options.requestName] = $.ajax(targetObj);
}

```